ui/src/Panels/Consumer/Consumer.view.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react/jsx-no-bind */
+import React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+
+import { translations } from './Consumer.assets.js';
+import { CONSTANTS, idAttributeGenerator } from 'Utils';
+import { Counter, ConsumerMessages } from 'Groups';
+import { ConsumerMessage } from 'Elements';
+import {
+  useTranslate,
+  useToggle,
+  useKafkaVertxWebSocket,
+} from 'ReactCustomHooks';
+import { Button } from 'carbon-components-react';
+
+const Consumer = (props) => {
+  const {
+    getConsumerWebsocket,
+    topic,
+    maxNumberOfMessages,
+    className,
+    ...others
+  } = props;
+  const classesToApply = clsx('Consumer', { [className]: className });
+
+  const translate = useTranslate(translations);
+  const [consumerRunning, toggleConsumerRunning] = useToggle(false);
+  const {
+    start,
+    stop,
+    isReady,
+    isRunning,
+    messages,
+    totalSuccessMessages,
+  } = useKafkaVertxWebSocket(getConsumerWebsocket, maxNumberOfMessages);
+
+  const onButtonClick = () => {
+    if (consumerRunning) {
+      stop();
+    } else {
+      start();
+    }
+    toggleConsumerRunning();
+  };
+
+  return (
+    <div {...others} className={classesToApply}>
+      <Counter
+        {...idAttributeGenerator('consumer_stats')}
+        title={translate('MESSAGES_CONSUMED', {}, true)}
+        subtitle={translate('FROM_TOPIC', { topic }, true)}
+        count={totalSuccessMessages}
+        className={'Consumer__count'}
+      />
+      <div className={'Consumer__control'}>
+        <Button
+          disabled={!isReady}
+          onClick={onButtonClick}
+          {...idAttributeGenerator('consumer_button')}
+        >
+          {isRunning
+            ? translate('STOP_CONSUMING')
+            : translate('START_CONSUMING')}
+        </Button>
+      </div>
+      <ConsumerMessages>
+        {messages.map((msg, index) => (
+          <ConsumerMessage
+            {...idAttributeGenerator('consumer_consumed_message')}
+            key={`consumed-message-${index}:${msg.index}`}
+            isFirst={index === 0}
+            error={
+              msg.status === CONSTANTS.VERTX_ERROR_STATUS
+                ? { message: translate('ERROR_CONSUMING', {}, true) }
+                : undefined
+            }
+            message={msg}
```

Comment:
This also causes PropType warnings when the message is an error. Same remedy options as before applies here.